File: api/src/main/java/io/perryquotes/api/quote/AuthorRepository.java
package io.perryquotes.api.quote;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface AuthorRepository extends JpaRepository<Author, UUID> {

  Optional<Author> findByUuid(UUID uuid);

  Optional<Author> findByName(String name);
}

File: core/src/main/kotlin/io/perryquotes/core/author/AuthorRepository.kt
package io.perryquotes.core.author

import io.perryquotes.core.jooq.tables.pojos.AuthorEntity
import java.util.*

interface AuthorRepository {

    fun findByUuid(uuid: UUID): AuthorEntity?
    fun findByName(name: String): AuthorEntity?
    fun findAll(): List<AuthorEntity>
    fun create(author: AuthorEntity): AuthorEntity
    fun update(author: AuthorEntity): AuthorEntity
    fun delete(uuid: UUID): Int
}
